tsconfig.json

```diff
@@ -1,6 +1,5 @@
 {
   "compilerOptions": {
-    "allowJs": true,
```

Comment:
The reason I avoided going full TypeScript is that we would have to install _a lot_ of typings packages that would only be used in that one file 😣

If it’s not too hard, and we don’t have to add too many packages, I’d be fine with going for it. Another thing to note is that a lot of tests are written in plain JS and they would be _very_ hard to convert. I think you wouldn’t need to convert them because they are executed by Jest. You may just need to add a custom transformer for Jest and test `.js` files.

Another alternative is we could just provide declarations for our exports, which while duplicative is a valid approach (their equivalence can be tested). Or do you want declarations for internal modules as well?